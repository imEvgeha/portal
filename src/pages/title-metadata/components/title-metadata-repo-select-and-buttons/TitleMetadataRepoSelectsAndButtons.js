import React from 'react';
import PropTypes from 'prop-types';
import IconActionAdd from '@vubiquity-nexus/portal-assets/icon-action-add.svg';
import NexusSavedTableDropdown from '@vubiquity-nexus/portal-ui/lib/elements/nexus-saved-table-dropdown/NexusSavedTableDropdown';
import {Button} from 'primereact/button';
import './TitleMetadataRepoSelectsAndButtons.scss';
import {TABLE_LABELS, TABLE_OPTIONS} from '../../constants';
import CatalogueOwner from '../catalogue-owner/CatalogueOwner';

const RepositorySelectsAndButtons = ({
    getNameOfCurrentTab,
    gridApi,
    columnApi,
    username,
    userDefinedGridStates,
    setUserDefinedGridState,
    applyPredefinedTableView,
    lastStoredFilter,
    setBlockLastFilter,
    changeCatalogueOwner,
    setShowModal,
}) => {
    if (getNameOfCurrentTab() === 'repository') {
        return (
            <div className="row nexus-c-title-metadata__select-container">
                <div className="d-flex justify-content-end col-xs-12 col-xl-10 col-xxl-11">
                    <NexusSavedTableDropdown
                        gridApi={gridApi}
                        columnApi={columnApi}
                        username={username}
                        userDefinedGridStates={userDefinedGridStates}
                        setUserDefinedGridState={setUserDefinedGridState}
                        applyPredefinedTableView={applyPredefinedTableView}
                        tableLabels={TABLE_LABELS}
                        tableOptions={TABLE_OPTIONS}
                        lastStoredFilter={lastStoredFilter}
                        setBlockLastFilter={setBlockLastFilter}
                        isTitleMetadata={true}
                    />
                    <CatalogueOwner setCatalogueOwner={changeCatalogueOwner} />
                </div>
                <div className="d-flex align-items-center justify-content-end col-xs-12 col-xl-2 col-xxl-1">
                    <Button
                        tooltip="Create New Title"
                        tooltipOptions={{position: 'left'}}
                        icon={IconActionAdd}
                        onClick={() => setShowModal(true)}
                        className="p-button-text nexus-c-title-metadata__create-btn"
                    />
                </div>
            </div>
        );
    }

    return null;
};

RepositorySelectsAndButtons.propTypes = {
    getNameOfCurrentTab: PropTypes.func,
    gridApi: PropTypes.any,
    columnApi: PropTypes.any,
    username: PropTypes.string,
    userDefinedGridStates: PropTypes.array,
    setUserDefinedGridState: PropTypes.func,
    applyPredefinedTableView: PropTypes.func,
    lastStoredFilter: PropTypes.object,
    setBlockLastFilter: PropTypes.func,
    changeCatalogueOwner: PropTypes.func,
    setShowModal: PropTypes.func,
};

RepositorySelectsAndButtons.defaultProps = {
    getNameOfCurrentTab: () => null,
    gridApi: null,
    columnApi: null,
    username: '',
    userDefinedGridStates: [],
    setUserDefinedGridState: () => null,
    applyPredefinedTableView: () => null,
    lastStoredFilter: {},
    setBlockLastFilter: () => null,
    changeCatalogueOwner: () => null,
    setShowModal: () => null,
};

export default RepositorySelectsAndButtons;
