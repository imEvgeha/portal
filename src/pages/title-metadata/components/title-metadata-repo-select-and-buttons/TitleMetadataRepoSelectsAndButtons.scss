.nexus-c-title-metadata__load-container {
  display: flex;
  justify-content: end;
  min-height: 52px;
}

.nexus-c-title-metadata__select-container {
  display: flex;
  justify-content: flex-end;
  padding-right: 20px;
}