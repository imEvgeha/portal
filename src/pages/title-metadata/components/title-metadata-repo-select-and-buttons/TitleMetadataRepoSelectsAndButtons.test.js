import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import {RepositorySelectsAndButtons} from './TitleMetadataRepoSelectsAndButtons';

describe('RepositorySelectsAndButtons', () => {
    let wrapper = null;
    let createBtn = null;
    const mockStore = configureStore();
    const store = mockStore({
        titleMetadata: {
            filter: {
                filterModel: {},
                sortModel: null,
                columnState: [],
            },
        },
    });

    beforeAll(() => {
        wrapper = shallow(
            <RepositorySelectsAndButtons
                getNameOfCurrentTab={() => 'repository'}
                store={store}
                currentUserView={undefined}
                setCurrentUserView={() => null}
            />
        );
        createBtn = wrapper.find('.nexus-c-title-metadata__create-btn');
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render title metadata repository and buttons', () => {
        expect(wrapper.find('.nexus-c-title-metadata__select-container')).toHaveLength(1);
    });

    it('should render Create button', () => {
        expect(createBtn.length).toEqual(1);
    });
});
