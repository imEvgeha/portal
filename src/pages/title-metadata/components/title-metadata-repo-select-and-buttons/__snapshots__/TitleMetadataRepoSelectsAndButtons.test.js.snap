// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleMetadataRepoSelectsAndButtons should match snapshot 1`] = `
<Row
  className="nexus-c-title-metadata__select-container"
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Col
    className="d-flex justify-content-end"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs={5}
  >
    <NexusSavedTableDropdown
      applyPredefinedTableView={[Function]}
      columnApi={null}
      externalFilter={Object {}}
      gridApi={null}
      isTitleMetadata={true}
      lastStoredFilter={Object {}}
      onUserDefinedViewSelected={[Function]}
      setBlockLastFilter={[Function]}
      setUserDefinedGridState={[Function]}
      tableLabels={
        Object {
          "savedDropdownLabel": "Saved Table View:",
          "savedViewslabel": "My Saved Views",
        }
      }
      tableOptions={
        Array [
          Object {
            "label": "All",
            "value": "all",
          },
        ]
      }
      userDefinedGridStates={Array []}
      username=""
    />
  </Col>
  <Col
    className="d-flex justify-content-end"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs={5}
  >
    <CatalogueOwner
      setCatalogueOwner={[Function]}
    />
  </Col>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs={2}
  >
    <ForwardRef
      className="p-button-text nexus-c-title-metadata__create-btn"
      icon={
        Object {
          "process": [Function],
        }
      }
      onClick={[Function]}
      tooltip="Create New Title"
      tooltipOptions={
        Object {
          "position": "left",
        }
      }
    />
  </Col>
</Row>
`;
